File: server/app/models/summary.py
from pydantic import BaseModel, Field


class ActionItem(BaseModel):
    task: str = Field(description="Việc cần làm, viết ngắn gọn")
    owner: str | None = Field(
        default=None,
        description="Người phụ trách nếu transcript nhắc tới rõ ràng, ngược lại để null",
    )
    due: str | None = Field(
        default=None,
        description="Thời hạn nếu transcript nhắc tới, giữ nguyên cách diễn đạt gốc",
    )


class MeetingSummary(BaseModel):
    summary: str = Field(description="Tóm tắt ngắn gọn nội dung cuộc họp (3-6 câu)")
    key_points: list[str] = Field(default_factory=list, description="Các ý chính đã thảo luận")
    action_items: list[ActionItem] = Field(default_factory=list)
    decisions: list[str] = Field(default_factory=list, description="Các quyết định chính đã đưa ra")

File: client/src/lib/summary.js
/**
 * Chuyển bản tóm tắt (MeetingSummary) sang Markdown để sao chép: heading cho từng mục, bullet cho danh sách,
 * việc cần làm dạng checklist. Thứ tự và tên mục giống bảng tóm tắt trên UI và file xuất.
 */
export function summaryToMarkdown(summary, title) {
  const lines = []
  if (title) lines.push(`# ${title}`, '')

  lines.push('## Tóm tắt', summary.summary?.trim() ?? '', '')

  if (summary.key_points?.length) {
    lines.push('## Ý chính', ...summary.key_points.map((p) => `- ${p}`), '')
  }

  lines.push('## Việc cần làm')
  if (summary.action_items?.length) {
    for (const item of summary.action_items) {
      const meta = [item.owner && `Phụ trách: ${item.owner}`, item.due && `Hạn: ${item.due}`].filter(Boolean).join(' · ')
      lines.push(`- [ ] ${item.task}${meta ? ` — ${meta}` : ''}`)
    }
  } else {
    lines.push('Không có việc cần làm nào được nhắc tới.')
  }
  lines.push('')

  lines.push('## Quyết định')
  if (summary.decisions?.length) lines.push(...summary.decisions.map((d) => `- ${d}`))
  else lines.push('Chưa ghi nhận quyết định nào.')

  return lines.join('\n').trim() + '\n'
}
